### PR TITLE
Restructured build workflow & separated code base into individual packages

### PR DESCRIPTION
**Pull request by PatrickShaw**
**Note:** Lots of changes made so be on the lookout for bugs.

**How to I watch the code base now?**
Just run `yarn watch`. No need for `yarn watch:frontend` and `yarn watch:backend`

**Why separate the code base into packages?**
 - Not everyone's going to want to use all of `openapi-platform`. Individual packages means people can pick and choose.
 - I want to use certain parts of `openapi-platform` for my own projects :)
 - Encourages the decoupling of separate sections of the code base.
 - You'll find a lot of popular repositories follow the same architecture. Babel, jest and storybook are just a few examples.

**What's gulp for?**
It has a nice API for building complex build workflows.
E.g. You can run certain scripts in parallel or in series which is nice. 

**Why use browser-sync instead of webpack-dev-server**
Mainly because I found it easier to use with gulp. It also has a few features webpack-dev-server doesn't and I consider the API to be less confusing to use.

**What on earth are all the changes for?**
Most of it is moving files around, refactoring imports and changing READMEs.
The less trivial line changes are to do with how builds are handled now (quite dramatically different).

### CODE REVIEW

File: Gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
+/**
+ * Inspiration for this file taken from https://github.com/babel/babel/blob/master/Gulpfile.js
+ */
+const colors = require('colors');
+const { join, sep, resolve } = require('path');
+const { spawn } = require('child_process');
```
[the remark this answers is not in the document]
Comment (PatrickShaw) [reply Gulpfile.js]: Ah, cool, sounds good to me :)

File: build-packages/build-util/README.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Build utils
```
[the remark this answers is not in the document]
Comment (PatrickShaw) [reply build-packages/build-util/README.md]: Oh lol, I think it was left over from where I copied and pasted it from (took it from one of my repos). Let's just leave it in there, each package should have a README.md

File: babel.config.js [hunk 3]
```diff
@@ -1,79 +1,64 @@
...
+  });
 
-const frontendSettings = createBabelSettings({
-  envSettings: {
-    targets: {
-      browsers: ['last 2 versions'],
+  const frontendSettings = createBabelSettings({
+    envSettings: {
+      targets: {
+        browsers: ['edge >= 12'],
```
[the remark this answers is not in the document]
Comment (PatrickShaw) [reply babel.config.js]: Yep I'll replace it with yours

File: babel.config.js [hunk 4]
```diff
@@ -1,79 +1,64 @@
...
+  storyshotsSettings.plugins.push('require-context-hook');
 
-module.exports = api => {
-  const env = api.env();
   const config = {
     ...defaultSettings,
     overrides: [
       {
-        test: 'src/frontend',
+        test: './packages/frontend/src',
```
[the remark this answers is not in the document]
Comment (PatrickShaw) [reply babel.config.js]: No it only transpiles the things in the package itself. At the moment everything outside gets transpiled to be compatible with the currently installed node version.

I'm going to fix this in another PR; packages are usually transpiled into both es5-module and es6-module variants.

**Comment by calebj0seph** [comment]
Approved, maybe merge my PR first though.

File: babel.config.js [hunk 5]
```diff
@@ -94,7 +79,7 @@ module.exports = api => {
       aren't run through Webpack.
     */
     config.overrides.push({
-      test: ['.storybook', 'test/snapshots/storyshots.test.tsx'],
+      test: [],
```
Comment (PatrickShaw) [inline babel.config.js]: Annnnnd there's a bug :P

**Pull request merged**